src/dishka/registry.py

```diff
@@ -320,12 +321,12 @@ def _process_decorator(
             )
         scope = self.dependency_scopes[provides]
         registry = self.registries[scope]
-        undecorated_type = NewType(
+        undecorated_type = NewType(  # type: ignore[misc]
             f"{provides.type_hint.__name__}@{self.decorator_depth[provides]}",
-            decorator.provides.type_hint,
+            decorator.provides.type_hint,  # type: ignore[name-defined]
```

Comment:
why?

Reply:
error: Name "decorator.provides.type_hint" is not defined  [name-defined]

Reply:
sounds like a bug